packages/msgraph/src/csdl.tsx

```diff
@@ -0,0 +1,580 @@
+    default:
+      return true;
+  }
+}
+
+export const CadlProgramViewer: FunctionComponent<CadlProgramViewerProps> = ({ program }) => {
+  const root = program.checker!.getGlobalNamespaceType();
+  const excludedNamespaces = ["Cadl"];
+  const namespaces = expandNamespaces(root)
+    .map((namespace) => ({
+      name: program.checker!.getNamespaceString(namespace),
+      namespace,
+    }))
+    .filter(
+      (ns) =>
+        ns.name &&
+        hasChildren(ns.namespace) &&
+        excludedNamespaces.every((exclude) => !ns.name.startsWith(exclude))
+    );
+  return (
+    <ProgramContext.Provider value={program}>
+      {namespaces.map((ns) => {
+        return (
+          <SchemaElement Namespace={ns.name}>
+            <Elements namespace={ns.namespace} />
+          </SchemaElement>
+        );
+      })}
+    </ProgramContext.Provider>
+  );
+};
+
+const Elements: FunctionComponent<{ namespace: NamespaceType }> = ({ namespace }) => {
+  return (
+    <>
+      <EnumTypeList enums={namespace.enums} />
+      <ModelList models={namespace.models} />
+      {/* <ItemList items={namespace.interfaces} render={(x) => <Interface type={x} />} />
+      <ItemList items={namespace.operations} render={(x) => <Operation type={x} />} />
+      <ItemList items={namespace.unions} render={(x) => <Union type={x} />} /> */}
+    </>
+  );
+};
+
+const EnumTypeList: FunctionComponent<{ enums: Map<string, EnumType> }> = (props) => (
+  <ItemList
+    items={props.enums}
+    render={(type) => (
+      <EnumTypeElement Name={type.name}>
+        <EnumMemberList members={type.members} />
+      </EnumTypeElement>
+    )}
+  />
+);
+
+const EnumMemberList: FunctionComponent<{ members: EnumMemberType[] }> = (props) => (
+  <ItemList
+    items={props.members}
+    render={(member) => <EnumMemberElement Name={member.name} Value={member.value} />}
+  />
+);
+
+const ModelList: FunctionComponent<{ models: Map<string, ModelType> }> = (props) => {
+  return (
+    <ItemList
+      items={props.models}
+      render={(model) => {
+        const program = useContext(ProgramContext);
+        const idProp = [...model.properties.entries()].find(([k, v]) => isId(program, v));
+        const isOpen = isOpenModel(program, model);
+        if (idProp) {
+          return (
+            <EntityTypeElement
+              Name={model.name}
+              BaseType={model.baseModel?.name}
+              {...(isOpen ? { OpenType: "true" } : {})}
+            >
+              {idProp && (
+                <KeyElement>
+                  <PropertyRefElement Name={idProp[1].name} />
+                </KeyElement>
+              )}
+              <PropertyList properties={model.properties} />
+            </EntityTypeElement>
+          );
+        } else {
+          return (
+            <ComplexTypeElement
+              Name={model.name}
+            >
+              <PropertyList properties={model.properties} />
+            </ComplexTypeElement>
+          );
+        }
```

Comment:
```suggestion
        }

        return (
          <ComplexTypeElement Name={model.name}>
            <PropertyList properties={model.properties} />
          </ComplexTypeElement>
        );
```

The else may not be necessary here
Formatting document should also have removed the extra newlines from the `ComplexTypeElement` definition.